app/Http/Controllers/Api/AssetsController.php

```diff
@@ -506,6 +520,13 @@ public function update(Request $request, $id)
             ($request->filled('company_id')) ?
                 $asset->company_id = Company::getIdForCurrentUser($request->get('company_id')) : '';
 
+            if ($request->has('image_source')) {
+                try {
```

Comment:
Hi there - what's the reason for a try/catch here?

Reply:
Hi, added a comment on that. Basically the called function can throw an error if source is not in supported format or somehow malformed.

Reply:
Hi - What's the error it can throw? It looks like you're just setting the $asset->image value, no? 

`$asset->image = $this->image($request->input('image_source'));`

Reply:
`$this->image()` can throw the exception

Reply:
Oh, sorry... I had no idea it works like that. As you might imagine I'm new to php. Removed both try/catches.